wgpu-hal/src/dx12/suballocation.rs

```diff
@@ -8,6 +8,9 @@ use committed as allocation;
 #[cfg(feature = "windows_rs")]
 use placed as allocation;
 
+// TODO: windows_rs is the default now, but we cannot actually remove the slow path as it is
+// used in a workaround for Intel Xe?  (without updating any of the comments/documentation)
+
 // This exists to work around https://github.com/gfx-rs/wgpu/issues/3207
 // Currently this will work the older, slower way if the windows_rs feature is disabled,
 // and will use the fast path of suballocating buffers and textures using gpu_allocator if
```

Comment:
@Elabajaba @xiaopengli89 is https://github.com/gfx-rs/wgpu/pull/3668 still a problem on Intel Xe? I'd like to clean up the now-meaningless `cfg(windows_rs)` feature, but probably have to keep `mod committed` around for this Xe workaround?

Reply:
@MarijnS95 This seems to be a driver issue, and it would be best to keep a workaround via options.

Reply:
This workaround has since been refactored to not be as intrusive to the implementation. Note that `gpu-allocator` has both committed and placed resource support in its API [^1].

I'm still a bit surprised by this feature from @Elabajaba, because we've found that on some hardware and types of textures, placed resources are much slower, because the hardware doesn't have enough knowledge to put them in the right space of memory. If anything both committed and placed resources should be used throughout the codebase.

[^1]: I don't think there was much use in having this feature _directly inside `gpu-allocator`_ since committed resources heavily clobber/confuse our API without doing _anything_ on the allocated blocks (it's just a call to the `ID3D12Device`). The implementation does however count this memory usage towards a "compatible" memory type which might help @nical with the counters.

Reply:
@xiaopengli89 I'd love to know if the issue on Xe can be reproduced and how, your issue from https://github.com/gfx-rs/wgpu/issues/3552 doesn't contain easily-runnable sample code to demonstrate the problem.

I'd love to see if it can be solved by _only_ allocating render targets with committed resources, and keep the rest placed on the heap.

Reply:
> This workaround has since been refactored to not be as intrusive to the implementation. Note that `gpu-allocator` has both committed and placed resource support in its API [1](#user-content-fn-1-a4aaadbeda1e73fb5f4201416273c529).
> 
> I'm still a bit surprised by this feature from @Elabajaba, because we've found that on some hardware and types of textures, placed resources are much slower, because the hardware doesn't have enough knowledge to put them in the right space of memory. If anything both committed and placed resources should be used throughout the codebase.

I'm not a graphics programmer and didn't know that (my entire knowledge of gpu memory allocation is fuzzy memories of some asawicki blog posts). Also I don't think the webgpu api gives you that level of control to choose what type of allocation you want (and dx12 seems to make this a lot more painful than vulkan, where you can query what allocation type a resource would prefer).

The feature gating was just the simplest way I could come up with so that Mozilla wouldn't have to vendor windows-rs at the time, while also vastly improving performance vs only committed resources.

Reply:
Yeah, on Vulkan there's an API that specifies if a dedicated allocation (synonimous to Dx12 committed resources) is preferred or even required. I don't know if `wgpu` and `gpu-alloc` take care of this, at least our `gpu-allocator` crate does.

You might have been reading https://www.asawicki.info/articles/memory_management_vulkan_direct3d_12.php even though it doesn't explicitly seem to list that placed resources are always better, but it becomes obvious that they allow for much finer grained placement and thus less wasted memory and likely also (opaque to us) extra driver overhead.

There's an old quote from an Nvidia [DX12 Do's And Don'ts](https://developer.nvidia.com/dx12-dos-and-donts) blog that is no longer available where it was stated that committed resources are always better. A newer post says that this is only the case for ucompressed resources _when aliasing_: https://developer.nvidia.com/blog/advanced-api-performance-memory-and-resources/